packages/channel-client/src/types.ts

```diff
@@ -69,3 +73,148 @@ export interface ChannelClientInterface<Payload = object> {
   pushMessage: (message: Message<Payload>) => Promise<PushMessageResult>;
   getAddress: () => Promise<string>;
 }
+
+export interface EventsWithArgs {
+  MessageQueued: [Message<ChannelResult>];
+  ChannelUpdated: [ChannelResult];
+  // TODO: Is `ChannelResult` the right type to use here?
+  ChannelProposed: [ChannelResult];
+}
+
+type UnsubscribeFunction = () => void;
+
+interface CreateChannelParameters {
+  participants: Participant[];
+  allocations: Allocation[];
+  appDefinition: string;
+  appData: string;
+}
+
+interface UpdateChannelParameters {
+  channelId: string;
+  participants: Participant[];
+  allocations: Allocation[];
+  appData: string;
+}
+
+export type GetAddressRequest = JsonRpcRequest<'GetAddress', {}>; // todo: what are params
+
+export type CreateChannelRequest = JsonRpcRequest<'CreateChannel', CreateChannelParameters>;
+
+export type CreateChannelResponse = JsonRpcResponse<ChannelResult>;
+
+export interface JoinChannelParameters {
+  channelId: string;
+}
+
+export type JoinChannelRequest = JsonRpcRequest<'JoinChannel', JoinChannelParameters>;
+
+export type JoinChannelResponse = JsonRpcResponse<ChannelResult>;
+
+export type UpdateChannelRequest = JsonRpcRequest<'UpdateChannel', UpdateChannelParameters>;
+
+export type PushMessageRequest = JsonRpcRequest<'PushMessage', Message>;
+
+export interface PushMessageResult {
+  success: boolean;
+}
+
+export type PushMessageResponse = JsonRpcResponse<PushMessageResult>;
+
+export interface CloseChannelParameters {
+  channelId: string;
+}
+export type CloseChannelRequest = JsonRpcRequest<'CloseChannel', CloseChannelParameters>;
+export type CloseChannelResponse = JsonRpcResponse<ChannelResult>;
+
+export type ChannelProposedNotification = JsonRpcNotification<'ChannelProposed', ChannelResult>;
+export type ChannelUpdatedNotification = JsonRpcNotification<'ChannelUpdated', ChannelResult>;
+export type ChannelClosingNotification = JsonRpcNotification<'ChannelClosed', ChannelResult>;
+
+export type MessageQueuedNotification = JsonRpcNotification<'MessageQueued', Message>;
+
+export type Notification =
+  | ChannelProposedNotification
+  | ChannelUpdatedNotification
+  | ChannelClosingNotification
+  | MessageQueuedNotification;
+
+export type NotificationName = Notification['method'];
+
+export type Request =
+  | GetAddressRequest
+  | CreateChannelRequest
+  | JoinChannelRequest
+  | UpdateChannelRequest
+  | PushMessageRequest
+  | CloseChannelRequest;
+
+export enum ErrorCodes {
+  SIGNING_ADDRESS_NOT_FOUND = 1000,
+  INVALID_APP_DEFINITION = 1001,
+  INVALID_APP_DATA = 1002,
+  UNSUPPORTED_TOKEN = 1003,
+  CHANNEL_NOT_FOUND = 1004
+}
+
+export class ChannelClientError implements JsonRpcErrorResponse {
```

Comment:
This and the other Error classes probably don't belong in types. They could probably be removed for the time being, as I don't think they're used anywhere.